qml/barV/BarV/Bars.qml

```diff
@@ -0,0 +1,38 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+
+Column {
+    spacing: 10
+
+    Repeater {
+        property var percentages: [Math.random(), Math.random(), Math.random(
+                ), Math.random(), Math.random()]
+        model: 5
+        delegate: Bar {
+            width: parent.width
+            height: 40
+            property real currentPercentage: Math.random()
+
+            Behavior on currentPercentage {
+                NumberAnimation {
+                    duration: 500
+                }
+            }
+
+            Timer {
+                interval: Math.random() * 2000 + 500
+                running: true
+                repeat: true
+                onTriggered: {
+                    var increment = Math.random() * 0.2 - 0.1
+                    currentPercentage = Math.max(
+                                Math.min(currentPercentage + increment, 1), 0)
+                }
+            }
+
+            onCurrentPercentageChanged: {
+                percentage = currentPercentage
+            }
```

Comment:
Non c'è bisogno di tutto questo: il componente `Bar` ha già una proprietà che si chiama `percentage`. Puoi assegnarla direttamente così:

```Qml
        delegate: Bar {
            width: parent.width
            height: 40
            percentage: percentages[index]
        }
```

Questo funziona perché quando utilizzi un `Repeater` passando un numero come `model` (in realtà a `model` potresti passare anche altre cose di cui qui non parleremo), la variabile `index` contiene l'indice dell'elemento corrente. In questo caso la possiamo usare per accedere l'elemento i-esimo dell'array.

Puoi togliere sia la proprietà `currentPercentage` che l'handler `onCurrentPercentageChanged`.